accessories/climate.js

```diff
@@ -99,6 +103,10 @@ HomeAssistantClimate.prototype = {
     serviceData.entity_id = this.entity_id;
     serviceData.temperature = value;
 
+    if (getTempUnits(this.data) === 'FAHRENHEIT') {
+      serviceData.temperature = celsiusToFahrenheit(serviceData.temperature);
+    }
+  
```

Comment:
I am not sure how this is really expected to work, especially since I am testing this code out and it is not working. HA will still understand that the thermostat is configured as F so it will say the converted value is too low, i.e. 60 F to C is 16, so HA will say it is too low to set.

Reply:
Been a while since I wrote this so I'm having trouble remembering some details... But if I recall correctly-- HomeKit only speaks in Celsius so when you ask HomeAssistant to change to a temperature in Fahrenheit then you need to convert back to C in order for HomeKit to understand it.

Reply:
To explain the code a little more clearly -- 

- if HomeAssistant temp units is Fahrenheit
- assume that temp being requested from user is in F
- convert serverData temperature (being sent to HomeKit) to Celsius
- send converted temp to HomeKit

Reply:
Correct, HomeKit seems to require C, but the issue is on the HA side, which I can tell from the logs from home bridge; it is essentially blocking home bridge from sending the value since it thinks the thermostat is set to F, although the number getting sent over is C. In other works, HA reads the thermostat as F, so a converted value like 60 F to 16 F is too low to set.

Reply:
May just need to flop the direction (C to F/F to C) in get and set?  If HA is in F and HK is in C, data going from HA to HK needs a F to C, and HK to HA needs a C to F.  I wrote a concept code for this that I think chris adapted, and may have misread the source/dest devices in get/set.

Reply:
Not sure but I tested this code on my setup at home and it worked perfect.

- HomeKit configured for F
- Nest configured for F
- (some background interactions happening in C, but not by my control)